doc/modules/linear_model.rst

```diff
@@ -850,28 +850,88 @@ regularization.
+As an optimization problem, binary
+class logistic regression with regularization term :math:`r(w)`  minimizes the
+following cost function:
 
-Elastic-Net regularization is a combination of :math:`\ell_1` and
-:math:`\ell_2`, and minimizes the following cost function:
+.. math:: \min_{w} C \sum_{i=1}^n \left(-y_i \log(\hat{p}(X_i)) - (1 - y_i) \log(1 - \hat{p}(X_i))\right) + r(w).
 
-.. math:: \min_{w, c} \frac{1 - \rho}{2}w^T w + \rho \|w\|_1 + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1),
 
-where :math:`\rho` controls the strength of :math:`\ell_1` regularization vs.
-:math:`\ell_2` regularization (it corresponds to the `l1_ratio` parameter).
+We currently provide four choices for the regularization term  :math:`r(w)`  via
+the `penalty` argument:
 
-Note that, in this notation, it's assumed that the target :math:`y_i` takes
-values in the set :math:`{-1, 1}` at trial :math:`i`. We can also see that
-Elastic-Net is equivalent to :math:`\ell_1` when :math:`\rho = 1` and equivalent
-to :math:`\ell_2` when :math:`\rho=0`.
++----------------+-------------------------------------------------+
+| penalty        | :math:`r(w)`                                    |
++================+=================================================+
+| `None`         | :math:`0`                                       |
++----------------+-------------------------------------------------+
+| :math:`\ell_1` | :math:`\|w\|_1`                                 |
++----------------+-------------------------------------------------+
+| :math:`\ell_2` | :math:`\frac{1}{2}\|w\|_2^2 = \frac{1}{2}w^T w` |
++----------------+-------------------------------------------------+
+| `ElasticNet`   | :math:`\frac{1 - \rho}{2}w^T w + \rho \|w\|_1`  |
++----------------+-------------------------------------------------+
+
+For ElasticNet, :math:`\rho` (which corresponds to the `l1_ratio` parameter)
+controls the strength of :math:`\ell_1` regularization vs. :math:`\ell_2`
+regularization. Elastic-Net is equivalent to :math:`\ell_1` when
+:math:`\rho = 1` and equivalent to :math:`\ell_2` when :math:`\rho=0`.
+
+Multinomial Case
+----------------
+
+The binary case can be extended to :math:`K` classes leading to the multinomial
+logistic regression, see also `log-linear model
+<https://en.wikipedia.org/wiki/Multinomial_logistic_regression#As_a_log-linear_model>`_.
+
+.. note::
+   It is possible to parameterize a :math:`K`-class classification model
+   using only :math:`K-1` weight vectors, leaving one class probability fully
+   determined by the other class probabilities by leveraging the fact that all
+   class probabilities must sum to one. We deliberately choose to overparameterize the model
+   using :math:`K` weight vectors for ease of implementation and to preserve the
+   symmetrical inductive bias regarding ordering of classes, see [16]_. This effect becomes
+   especially important when using regularization.
```

Comment:
Maybe we should add that this is a bad choice for unpenalized models!

Reply:
Added a small addition to the note